### ISSUE TITLE
Question regarding HTTPS listeners

### ISSUE THREAD

**Issue by root-io** [envoyproxy/gateway#2159]
*Description*:
I am having issue using Gateway, I've bee using version from v0.5.0 to latest, today I tried v0.6.0 but still not able to make it work.

My services are accessible with HTTP but not HTTPS.

The same exact config works days/weeks ago using latest but I had CORS errors, then realized it was not CORS because my services returns correct CORS headers but it was because the endpoint was offline randomly. Am I doing something wrong? Thanks!

```yaml
apiVersion: gateway.networking.k8s.io/v1
kind: Gateway
metadata:
  name: eg
  annotations:
    cert-manager.io/cluster-issuer: letsencrypt
    cert-manager.io/common-name: example.com
spec:
  gatewayClassName: eg
  listeners:
    - name: http-web
      protocol: HTTP
      hostname: example.com
      port: 80
    - name: http-subdomain1
      protocol: HTTP
      hostname: subdomain1.example.com
      port: 80
    - name: http-subdomain1-alias
      protocol: HTTP
      hostname: subdomain1-alias.example.com
      port: 80
    - name: http-subdomain2
      protocol: HTTP
      hostname: subdomain2.example.com
      port: 80
    - name: https-web
      protocol: HTTPS
      port: 443
      hostname: example.com
      tls:
        certificateRefs:
        - kind: Secret
          group: ""
          name: eg-https
    - name: https-subdomain1
      protocol: HTTPS
      port: 443
      hostname: subdomain1.example.com
      tls:
        certificateRefs:
        - kind: Secret
          group: ""
          name: eg-https
    - name: https-subdomain1-alias
      protocol: HTTPS
      port: 443
      hostname: subdomain1-alias.example.com
      tls:
        certificateRefs:
        - kind: Secret
          group: ""
          name: eg-https
    - name: https-subdomain2
      protocol: HTTPS
      port: 443
      hostname: subdomain2.example.com
      tls:
        certificateRefs:
        - kind: Secret
          group: ""
          name: eg-https
---
apiVersion: gateway.networking.k8s.io/v1
kind: HTTPRoute
metadata:
  name: foo
spec:
  parentRefs:
    - name: eg
  rules:
    - backendRefs:
        - group: ""
          kind: Service
          name: foo
          port: 3000
      matches:
        - path:
            type: PathPrefix
            value: /
    - filters:
        - type: RequestRedirect
          requestRedirect:
            scheme: https
            statusCode: 301
            hostname: subdomain1.example.com
            port: 443
  hostnames:
    - subdomain1.example.com
    - subdomain1-alias.example.com
---
apiVersion: v1
kind: Service
metadata:
  name: foo
  labels:
    app: foo
    service: foo
spec:
  ports:
    - name: http
      port: 3000
      targetPort: 3000
  selector:
    app: foo
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: foo
spec:
  replicas: 1
  selector:
    matchLabels:
      app: foo
      version: v1
  template:
    metadata:
      labels:
        app: foo
        version: v1
    spec:
      serviceAccountName: foo
      containers:
        - image: example/foo
          name: foo
          ports:
            - containerPort: 3000
```

**Comment by ionutvilie** [comment]
@arkodg  I ended up here searching for https in the issues. At first, this infinite loop looked like a bug but apparently, it is the desired behavior and documented in the API specs.

I have found a working example:

- one HTTPRoute is bound to the HTTP listener by configuring the `sectionName` in `spec.parentRefs`
https://github.com/kubernetes-sigs/gateway-api/blob/main/examples/standard/http-redirect.yaml#L45
- the second HTTPRoute must be bound to the HTTPS listener using the same configuration otherwise it won't work.
https://github.com/kubernetes-sigs/gateway-api/blob/main/examples/standard/http-redirect.yaml#L62


My working example.

```yaml
apiVersion: gateway.networking.k8s.io/v1beta1
kind: HTTPRoute
metadata:
  name: tls-redirect
  namespace: prod
spec:
  parentRefs:
    - kind: Gateway
      name: envoy-gateway
      namespace: kube-public
      sectionName: http
  hostnames:
    - "*.ak1pro.n1l.ro"
  rules:
    - filters:
        - type: RequestRedirect
          requestRedirect:
            scheme: https
---
apiVersion: gateway.networking.k8s.io/v1beta1
kind: HTTPRoute
metadata:
  name: homarr-prod
  namespace: prod
spec:
  parentRefs:
    - kind: Gateway
      name: envoy-gateway
      namespace: kube-public
      sectionName: https
  hostnames:
    - "homarr.ak1pro.n1l.ro"
  rules:
    - matches:
        - path:
            type: PathPrefix
            value: /
      backendRefs:
        - name: homarr
          port: 80
```

**Comment by root-io** [comment]
Thanks for the help, the redirection now works correctly but I still get ERR_CONNECTION_REFUSED on HTTPS, I think it was working on some commit after v0.5.0 and before v0.6.0-rc1 but I'm struggling to reproduce. Using cert-manager v1.13.2

**Comment by ionutvilie** [comment]
@root-io see if this helps, it works for me. I deployed the controller to kube-public 
https://github.com/ionutvilie/awesome-home-server/blob/master/ak1pro/kube-public/readme.md

Service-specific HTTP routing is present in the stage Kustomization: 
https://github.com/ionutvilie/awesome-home-server/blob/master/ak1pro/prod/httproutes.yaml

From what you are describing it seems like the gateway is not binding on the expected TCP/IP Address (LoadBalancer service).

**Comment by arkodg** [comment]
hey @ionutvilie thanks for creating the example,can you help update our  https://gateway.envoyproxy.io/v0.6.0/user/http-redirect/ docs with a PR to show the http->https redirect case (might also need some extra steps to generate certs which is present [here](https://gateway.envoyproxy.io/v0.6.0/user/secure-gateways/))

**Comment by root-io** [comment]
@ionutvilie Are you sure you are using v0.6.0, your yaml are using `gateway.networking.k8s.io/v1beta1` not `gateway.networking.k8s.io/v1`?

**Comment by ionutvilie** [comment]
hi @root-io, I updated the PR, as per my home server configuration, if the API did not warn me I did not notice. :)

I think I blindly copied the examples from the documentation and most of them are with `v1beta1 `. Atm I'm confused, I'm waiting for a review. I did not had time to figure out how the documentation is structured. Is it ok to change to v1 in 0.6 release or just in latest? 

Anyway, the example is tested and it works, hugo runs ok in dev mode.
